packages/clerk-js/src/ui/elements/Select.tsx

```diff
@@ -0,0 +1,316 @@
+          </SelectOptionsContainer>
+        </Flex>
+      )}
+    </>
+  );
+};
+
+type SelectOptionBuilderProps<T> = {
+  option: T;
+  index: number;
+  optionBuilder: OptionBuilder<T>;
+  handleSelect: (option: T) => void;
+  isSelected: boolean;
+};
+
+const _SelectOptionBuilder = <T extends DefaultOption>(
+  props: SelectOptionBuilderProps<T>,
+  ref?: React.ForwardedRef<HTMLDivElement>,
+) => {
+  const { option, optionBuilder, index, handleSelect, isSelected } = props;
+  return (
+    <Flex
+      ref={ref}
+      sx={{
+        userSelect: 'none',
+      }}
+      onClick={() => {
+        handleSelect(option);
+      }}
+    >
+      {optionBuilder(option, index, isSelected)}
+    </Flex>
+  );
+};
+
+const SelectOptionBuilder = React.memo(React.forwardRef(_SelectOptionBuilder)) as <T extends DefaultOption>(
+  props: SelectOptionBuilderProps<T> & { ref?: React.ForwardedRef<HTMLDivElement> },
+) => ReturnType<typeof _SelectOptionBuilder>;
+
+const SelectSearchbar = (props: PropsOfComponent<typeof InputWithIcon>) => {
+  const { sx, ...rest } = props;
+  React.useEffect(() => {
+    // @ts-expect-error
+    return () => props.onChange({ target: { value: '' } });
+  }, []);
+
+  return (
+    <Flex sx={theme => ({ borderBottom: theme.borders.$normal, borderColor: theme.colors.$blackAlpha200 })}>
+      <InputWithIcon
+        focusRing={false}
+        autoFocus
+        leftIcon={
+          <Icon
+            colorScheme='neutral'
+            icon={MagnifyingGlass}
+          />
+        }
+        sx={[{ border: 'none', borderRadius: '0' }, sx]}
+        {...rest}
+      />
+    </Flex>
+  );
+};
+
+export const SelectNoResults = (props: PropsOfComponent<typeof Text>) => {
+  const { sx, ...rest } = props;
+  return (
+    <Text
+      as='div'
+      variant='smallRegular'
+      sx={[theme => ({ width: '100%', padding: `${theme.space.$2} 0 0 ${theme.space.$4}` }), sx]}
+      {...rest}
+    />
+  );
+};
+
+export const SelectOptionsContainer = (props: PropsOfComponent<typeof Flex>) => {
+  const { sx, ...rest } = props;
+
+  return (
+    <Flex
+      direction='col'
+      sx={[
+        theme => ({
+          overflowY: 'scroll',
+          maxHeight: '18vh',
+          paddingBottom: theme.space.$2,
+        }),
+        sx,
+      ]}
+      {...rest}
+    />
+  );
+};
+
+export const SelectTrigger = React.forwardRef<HTMLButtonElement, PropsOfComponent<typeof Button> & { isOpen: boolean }>(
+  (props, ref) => {
+    const { sx, ...rest } = props;
+    return (
+      <Button
+        ref={ref}
+        onClick={props.onClick}
+        colorScheme='neutral'
+        variant='ghost'
+        textVariant='smallMedium'
+        isDisabled={props.isDisabled}
+        sx={[
+          theme => ({
+            paddingLeft: theme.space.$3x5,
+            paddingRight: theme.space.$3x5,
+            backgroundColor: theme.colors.$blackAlpha50,
+            position: 'absolute',
```

Comment:
❓ Are we sure we need absolute positioning for the trigger button? I'd expect the opposite.

Reply:
You're right. That got away. This needs to be handled by the PhoneInput.